try.go

```diff
@@ -0,0 +1,46 @@
+package lo
+
+func Try(callback func() error) (err bool) {
```

Comment:
I'm not a big fan of the `Try` function.

IMO we should only provide `TryWithErrorValue`. Go developers are encouraged to not ignore errors.

If you really want so, you can still ignore the returned value with `_`.

Also, having `TryWithErrorValue` named `Try` will promote it as the by-default function.

WDYT ?

Reply:
I added it because I have this in some of my projects as a helper function. Yes I know, I shouldn't ignore error values but sometimes it's nicer to write 

```go
if !Try(someFunction) {
  panic("errorred!")
}
```

than 

```go
if err := someFunction(); err != nil {
  panic("errorred!")
} 
```

Reply:
Originally this was meant for functions that return a lot of values that I ignore and only need the error status from but I'm still playing around with that. So instead of 

```go
if err, _, _, _ := someFunction(); err != nil {
  panic("errorred!")
} 
```

I could write

```go
if !Try(someFunction) {
  panic("errorred!")
}
```

than

Reply:
Yes, I see.

Good point.

Reply:
I have done some refactoring. There are now some overload methods for Try

So for the case above (so 3 return values and one error value) you could just write

```go
if !Try4(someFunction) {
  panic("errorred!")
}
```

WDYT? @samber